framework/src/dirackernels/ConstantPointSource.C

```diff
@@ -17,39 +19,167 @@ InputParameters
 ConstantPointSource::validParams()
 {
   InputParameters params = DiracKernel::validParams();
-  params.addRequiredParam<Real>("value", "The value of the point source");
-  params.addRequiredParam<std::vector<Real>>("point", "The x,y,z coordinates of the point");
+
+  params.addClassDescription("Apply a constant point load.");
+
+  params.addParam<std::vector<Real>>("value", "The list of values (v0...vn) of the point sources");
+  params.addParam<std::vector<Real>>("point",
+                                     "The list of coordinates (x0,y0,z0...xn,yn,zn) of the points");
   params.declareControllable("value");
+  params.addParam<FileName>("position_value_file",
+                            "The file containing positions and corresponding values.");
+  params.addParam<VectorPostprocessorName>(
+      "vector_postprocessor",
+      "The name of the VectorPostprocessor containing positions and corresponding values with "
+      "columns named x y z value");
+
```

Comment:
I think this crams 3 objects worth of functionality into one object. I'd rather see this refactored into `FilePointSource` and `VectorPostprocessorPointSource`

Reply:
I'd begrudgingly accept the file reader option in Constant point source (there is precedent for this). But I don't really like it.

The VPP is a hard no for me, as that is not "constant" at all anymore.

Reply:
I reverted the vpp part of this commit.  It should be its own type of point source.  I had implemented it wrong and it would have been constant.  I need to make the point source change as the vpp changes which the previous commit did not do.
The file reader is still part of constantPointSource

Reply:
I made a new object called FileConstantPointSource and moved the file reading to that.  I reverted all of the changes I made to ConstantPointSource so constantPointSource will only read in a single point source per block.  I need to move the vpp part to another object but that isn't constant so I had implemented it wrong.